mitodl/micromasters
# Upgrade to Elasticsearch 6

#### What are the relevant tickets?

https://github.com/mitodl/micromasters/issues/3748
https://github.com/mitodl/salt-ops/issues/867

#### What's this PR do?

It revives the changes that were attempted in these commits:
8886a8d (Rename wildcard)
c7d5797 (Bump Elasticsearch Version)
... both of which were introduced in https://github.com/mitodl/micromasters/pull/4303 and reverted in https://github.com/mitodl/micromasters/pull/4319

In addition, it changes the Elasticsearch Index Type, per "[Schedule for Removal of Mapping Types](https://www.elastic.co/guide/en/elasticsearch/reference/6.6/removal-of-types.html)"

#### How should this be manually tested?

We need to make sure that the error reported in https://github.com/mitodl/micromasters/issues/4318 does not reoccur. The test suite did not catch this the last time around, and it is passing now for this branch.

I could use some guidance on adding a test to cover the query at https://github.com/mitodl/micromasters/blob/e890c10a1025fd5cdb450d48aa82a3d69314d318/search/api.py#L296 ... unless someone can update this PR documenting that this is unnecessary.

I'm reading `search.indexing_api.py`, specifically the [assignment of PERCOLATE_MAPPING](https://github.com/mitodl/micromasters/blob/c4de3969c5e66e9f884729896808114b3c9eb1d4/search/indexing_api.py#L222-L232) but it's not immediately apparent how this mapping is put into place. The indices that I'm seeing in my Elasticsearch Docker instance that were created during the test suite run do not have a `query` field.

#### Releasing

I'm flagging @noisecapella on this PR because he's the Release Manager next week. We'll need to coordinate before this is released and deployed to production, because I will first need to transfer the original Elasticsearch 5 indices to the Elasticsearch 6 cluster (to the `production-apps` cluster) and modify their mapping types in the process (from "doc" to "_doc"). Someone will then need to change the production app's configuration to point at the ES 6 cluster in sync with the deployment.

I believe that someone needs to run a Celery reindexing task after the deployment to pick up documents that may have been added to the indices since they was copied.
